vowpalwabbit/cb_adf.cc

```diff
@@ -288,18 +293,13 @@ example* test_adf_sequence(multi_ex& ec_seq)
   return ret;
 }
 
-template <bool is_learn>
-void cb_adf::do_actual_learning(multi_learner& base, multi_ex& ec_seq)
+void cb_adf::learn(multi_learner& base, multi_ex& ec_seq)
 {
   _offset = ec_seq[0]->ft_offset;
   _gen_cs.known_cost = get_observed_cost(ec_seq);  // need to set for test case
-  if (is_learn && test_adf_sequence(ec_seq) != nullptr)
+  if (test_adf_sequence(ec_seq) != nullptr)
```

Comment:
this refactor is not equivalent, not sure if it's by design. What happens if in the past you would have to support call of `is_learn = true and test_adf_sequence(ec_seq) == nullptr`. After this change they would produce different calls, since the new learn() does not have the else clause anymore.

Reply:
test_adf_sequence checks for the existence of a valid label.  This is necessary condition for all the learn functions in the switch statement below.

Correctness was verified by RunTests that exercises this code.